src/components/Clock.tsx

```diff
@@ -0,0 +1,58 @@
+import { Component } from 'react';
+
+type ClockProps = {
+  clockName: string,
+};
+
+type ClockState = {
+  date: Date,
+};
+
+export class Clock extends Component<ClockProps, ClockState> {
+  state = {
+    date: new Date(),
+  };
+
+  dateId = 0;
```

Comment:
```suggestion
  dateIntervalId = 0;
```
it's more descriptive